wbooks/src/app/index.tsx

```diff
@@ -8,20 +8,29 @@
  * @format
  */
 
+import 'react-native-gesture-handler';
 import React from 'react';
-import { SafeAreaView, StatusBar } from 'react-native';
+import { NavigationContainer } from '@react-navigation/native';
+import { createStackNavigator } from '@react-navigation/stack';
+import SCREENS from '@constants/screens';
+import { RootStackParamList } from '@interfaces/rootStackParamList';
+import BookDetails from '@screens/BookDetails';
 import Library from '@screens/Library';
 
-import styles from './styles';
+const Stack = createStackNavigator<RootStackParamList>();
 
 const App = () => {
   return (
-    <>
-      <StatusBar barStyle="dark-content" />
-      <SafeAreaView style={styles.safeAreaView}>
-        <Library />
-      </SafeAreaView>
-    </>
+    <NavigationContainer>
+      <Stack.Navigator initialRouteName={SCREENS.LIBRARY}>
+        <Stack.Screen name={SCREENS.LIBRARY} component={Library} />
+        <Stack.Screen
+          name={SCREENS.BOOK_DETAILS}
+          component={BookDetails}
+          options={{ title: 'Book Detail' }}
+        />
+      </Stack.Navigator>
+    </NavigationContainer>
```

Comment:
Create a new component for the `AppNavigator`:
```
  return (
    <NavigationContainer>
      <StatusBar barStyle="dark-content" />
      <AppNavigator />
    </NavigationContainer>
  );
```

And then, `src/app/components/AppNavigator`:
```
function AppNavigator() {
  return (
    <Stack.Navigator initialRouteName={SCREENS.LIBRARY}>
      <Stack.Screen name={SCREENS.LIBRARY} component={Library} />
      <Stack.Screen
        name={SCREENS.BOOK_DETAILS}
        component={BookDetails}
        options={{ title: 'Book Detail' }}
      />
    </Stack.Navigator>
  );
}
```

Reply:
Refactored